### PR TITLE
Check vertical extremes

### PR DESCRIPTION
**Pull request by Hoolean**
This branch contains:

* The script used to generate [our report on the y-axis extremes in each source](https://docs.google.com/spreadsheets/d/1ISNGNcIr-gFNhjCZUb2AmZ9N6aRNe2LQYQofAshdogw/edit?usp=sharing)
* A FontBakery check for this project's OS/2.WinAscent and OS/2.WinDescent semantics

The latter is WIP, and in general this branch is non-urgent and does not need merging until after the v2.003 release :)

### CODE REVIEW

**Comment by chrissimpkins** [comment]
Does this implementation cover *shaped* y-min/max extremes?

**Comment by chrissimpkins** [comment]
Seems ok to merge?  Is there anything outstanding in this PR at this stage?

File: qa/checks/googlesans.py [hunk 1]
```diff
@@ -298,3 +300,86 @@ def com_google_fonts_check_googlesansflex_variable_fvar_default(font: Font, ttFo
...
+        variated_tags = {
+            tag for record in mvar.table.ValueRecord for tag in record.ValueTag
+        }
+
+        if "hcla" in variated_tags:
+            yield FAIL, "TTF varies OS/2.usWinAscent across its designspace"
+        if "hcld" in variated_tags:
+            yield FAIL, "TTF varies OS/2.usWinDescent across its designspace"
+
+    # TODO: This is only named instances; we should include everywhere we have
```
Comment (RickyDaMa) [inline qa/checks/googlesans.py:347]: Still outstanding

File: qa/checks/googlesans.py [hunk 2]
```diff
@@ -298,3 +300,86 @@ def com_google_fonts_check_googlesansflex_variable_fvar_default(font: Font, ttFo
...
+            (_, y_min, _, y_max) = bounds_pen.bounds
+
+            smallest_y_min = (
+                y_min if smallest_y_min is None else min(y_min, smallest_y_min)
+            )
+            largest_y_max = (
+                y_max if largest_y_max is None else max(y_max, largest_y_max)
+            )
+
+    # TODO: Round floats in some direction (but consider how this will interact with the +1).
```
Comment (RickyDaMa) [inline qa/checks/googlesans.py:380]: Still outstanding

**Review by RickyDaMa — COMMENTED** [review]
These should probably be addressed before merging

Also, we may want to revise the approach (e.g. looking at shaped strings) based on the discoveries of the Pixel fallback stack saga

**Comment by MariannaPaszkowska** [comment]
@Hoolean "It's a good idea to wait till after the strategy for the Android subsets is defined."

**Comment by belluzj** [comment]
@chrissimpkins should we close this PR now that we've decided not to subset GS Flex? Thanks in advance

**Comment by belluzj** [comment]
Chris: not related to GSF subsetting, so we should keep.
Chris: intent for Win metrics: every time we add glyphs we should adjust the win metrics (the values are entered by hand). It's not a novel approach, we've been doing that for Google Sans and we should keep doing that for all the brand families

**Comment by RickyDaMa** [comment]
> Does this implementation cover _shaped_ y-min/max extremes?

No, it just checks glyphs individually

**Comment by RickyDaMa** [comment]
Approach taken in this PR to establish extremes is not the approach we want based on https://github.com/googlefonts/googlesans-code/issues/56#issuecomment-2531718958

Closing but not deleting branch as there may be some useful code here for future developments

**Pull request closed without merging**